include/vkwrap/vkwrap/shader_module.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include "common/vulkan_include.h"
+#include "utils/files.h"
+#include "vkwrap/core.h"
+
+namespace vkwrap
+{
+class ShaderModule : private vk::UniqueShaderModule
+{
+  public:
+    using BaseType = vk::UniqueShaderModule;
+    using BaseType::get;
+
+    operator vk::ShaderModule() const& { return get(); }
+
+    ShaderModule( const std::string& file_path, vk::Device device )
+        : BaseType()
+    {
+
+        auto code_buffer = utils::readFileRaw( file_path );
+
+        vk::ShaderModuleCreateInfo create_info{
+            .codeSize = code_buffer.size(),
+            .pCode = reinterpret_cast<const uint32_t*>( code_buffer.data() ) };
+
+        *static_cast<BaseType*>( this ) = device.createShaderModuleUnique( create_info );
+    }
```

Comment:
Maybe this code could be moved to a static member function returning ```BaseType``` like:

```cpp
private:
BaseType create(const std::filesystem::path& file_path, vk::Device device) 
{
  ....
  return device.createShaderModuleUnique( create_info );
}

....

public:

ShaderModule( const std::filesystem::path& file_path, vk::Device device )
        : BaseType{create(file_path, device)}
{
}
```

Reply:
Agree, it's better